StreamReader.cpp

```diff
@@ -37,34 +37,58 @@ void FileStreamReader::Seek(uint64_t position) {
 	file.seekg(position, std::ios_base::beg);
 }
 
-void FileStreamReader::SeekRelative(int offset) {
+void FileStreamReader::SeekRelative(int64_t offset) {
 	file.seekg(offset, std::ios_base::cur);
 }
 
 
-MemoryStreamReader::MemoryStreamReader(char* buffer, size_t size) {
+MemoryStreamReader::MemoryStreamReader(char* buffer, uint64_t size) {
 	streamBuffer = buffer;
 	streamSize = size;
-	offset = 0;
+	position = 0;
 }
 
-void MemoryStreamReader::Read(char* buffer, size_t size) 
+void MemoryStreamReader::Read(char* buffer, uint64_t size) 
 {
-	if (offset + size > streamSize) {
+	if (position + size > streamSize) {
 		throw std::runtime_error("Size of bytes to read exceeds remaining size of buffer.");
 	}
 
-	memcpy(buffer, streamBuffer + offset, size);
-	offset += size;
+	// std::memcpy takes a _Size to copy of size_t. 
+	// Depending on the system, size_t may not be able to store the max value of uint64_t. 
+	// SIZE_MAX is the largest value that may be stored in size_t for the given system.
```

Comment:
Very interesting comment.

The stream position may be 64-bit, though the size that can be read into memory at any one time will be limited by the address space, which is represented by `size_t`. It makes sense for size to be declared as `size_t` as this is memory space, while the file offset is 64-bit to ensure files greater than 4GB (or greater than the memory size) can be supported.